dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/All.java

```diff
@@ -118,43 +147,89 @@ protected void forkRepositoriesFound(Map<String, String> parentToPath,
+            try {
+                forkedRepo = dockerfileGitHubUtil.getRepo(currUserRepo.getFullName());
+            } catch (FileNotFoundException e) {
+                /* The edge case here: If a different command calls getGHRepositories, and then this command calls
+                 * it again within 60 seconds, it will still have the same list of repositories (because of caching).
+                 * However, between the previous and current call, if some of those repositories are deleted, the call
+                 * above may cause a FileNotFoundException. This clause prevents that exception from stopping our call;
+                 * we do not need to stop because getGHRepositories checks that we have all the repositories we need.
+                 *
+                 * The integration test calls the testParent -> testAllCommand -> testIdempotency, and the
+                 * testIdempotency was failing because of this edge condition.
+                 */
+
+                log.warn("This repository does not exist. The list of repositories must be outdated, but the list" +
+                        "contains the repositories we need, so we ignore this error.");
+                return;
+            }
         } else {
             return;
         }
-        GHRepository parent = retrievedRepo.getParent();
+        GHRepository parent = forkedRepo.getParent();
 
-        if (parent == null || !parentToPath.containsKey(parent.getFullName())) {
+        if (parent == null || !pathToDockerfilesInParentRepo.containsKey(parent.getFullName())) {
             return;
         }
-        log.info("Fixing Dockerfiles in {}...", initialRepo.getFullName());
-        String parentName = parent.getFullName();
-        String image = parentToImage.get(parentName);
-        String tag = imageToTagMap.get(image);
-        String branch = retrievedRepo.getDefaultBranch();
-        GHContent content = dockerfileGitHubUtil.tryRetrievingContent(retrievedRepo, parentToPath.get(parentName),
-                branch);
-        dockerfileGitHubUtil.modifyOnGithub(content, branch, image, tag, commitMessage);
-        dockerfileGitHubUtil.createPullReq(parent, branch, retrievedRepo, message);
-    }
 
+        log.info("Fixing Dockerfiles in {}...", forkedRepo.getFullName());
+        String parentName = parent.getFullName();
+        String branch = (ns.get("b") == null) ? forkedRepo.getDefaultBranch() : ns.get("b");
+
+        String pathToDockerfile;
+        String image;
+        String tag;
+        GHContent content;
+        boolean isContentModified = false;
+        boolean isRepoSkipped = true;
+
+        Iterator<String> pathToDockerfileInParentRepoIterator = pathToDockerfilesInParentRepo.get(parentName).iterator();
+        Iterator<String> imagesFoundInParentRepoIterator = imagesFoundInParentRepo.get(parentName).iterator();
+
+        while (pathToDockerfileInParentRepoIterator.hasNext() && imagesFoundInParentRepoIterator.hasNext()) {
```

Comment:
Are `pathToDockerfileInParentRepoIterator` and `imagesFoundInParentRepoIterator` supposed to be the same number of elements?

Reply:
Yes they store the same number of elements - [All.java#115](https://github.com/salesforce/dockerfile-image-update/pull/25/files#diff-850febb926ac48d58dc0962f42249092R115) and [All.java#116](https://github.com/salesforce/dockerfile-image-update/pull/25/files#diff-850febb926ac48d58dc0962f42249092R116).

Reply:
Why not iterate on only one of them then? You can make the object immutable after they are created.

Reply:
Hmm.. Now iterating over only one of them but I think ideally it should be a class that stores repoName with pathToDockerfiles and corresponding images in those Dockerfiles.